pkg/synthesis/k8sPolicies.go

```diff
@@ -204,9 +205,16 @@ func toSelector(con symbolicexpr.Conjunction) *meta.LabelSelector {
 	for _, a := range con {
 		if !a.IsTautology() {
 			label, notIn := a.AsSelector()
+			label = toLegalK8SString(label)
 			req := meta.LabelSelectorRequirement{Key: label, Operator: boolToOperator[notIn]}
 			selector.MatchExpressions = append(selector.MatchExpressions, req)
 		}
 	}
 	return selector
 }
+
+var reg = regexp.MustCompile(`[^-A-Za-z0-9_.]`)
+
+func toLegalK8SString(s string) string {
+	return reg.ReplaceAllString(s, "-NLC")
```

Comment:
can you explain/add comment what this does?

Reply:
added comment.
also open relevant issue